src/snowflake/snowpark/modin/plugin/_internal/binary_op_utils.py

```diff
@@ -312,314 +300,527 @@ def _op_is_between_timedelta_and_numeric(
+        datatype: DataTypeGetter,
+    ) -> "BinaryOp":
+        """
+        Compute the binary operation between a scalar and a Snowpark column.
+        Args:
+            op: the name of binary operation
+            first_operand: Scalar value
+            second_operand: The SnowparkColumn for rhs
+            datatype: Callable for Snowpark data type
+            it is not needed.
+
+        Returns:
+            SnowparkPandasColumn for translated pandas operation
+        """
+
+        def first_datatype() -> DataType:
+            return infer_object_type(first_operand)
+
+        return BinaryOp.create(
+            op, pandas_lit(first_operand), first_datatype, second_operand, datatype
         )
-    elif op in (
-        "eq",
-        "ne",
-        "gt",
-        "ge",
-        "lt",
-        "le",
-    ) and _op_is_between_two_timedeltas_or_timedelta_and_null(
-        first_datatype(), second_datatype()
-    ):
-        # These operations, when done between timedeltas, work without any
-        # extra handling in `snowpark_pandas_type` or `binary_op_result_column`.
+
+    def _custom_compute(self) -> None:
+        """Implement custom compute method if needed."""
         pass
-    elif op == "mul" and (
-        _op_is_between_timedelta_and_numeric(first_datatype, second_datatype)
-    ):
-        binary_op_result_column = cast(
-            floor(first_operand * second_operand), LongType()
-        )
-        snowpark_pandas_type = TimedeltaType()
-    # For `eq` and `ne`, note that Snowflake will consider 1 equal to
-    # Timedelta(1) because those two have the same representation in Snowflake,
-    # so we have to compare types in the client.
-    elif op == "eq" and (
-        _op_is_between_timedelta_and_numeric(first_datatype, second_datatype)
-    ):
-        binary_op_result_column = pandas_lit(False)
-    elif op == "ne" and _op_is_between_timedelta_and_numeric(
-        first_datatype, second_datatype
-    ):
-        binary_op_result_column = pandas_lit(True)
-    elif (
-        op in ("truediv", "floordiv")
-        and isinstance(first_datatype(), TimedeltaType)
-        and _is_numeric_non_timedelta_type(second_datatype())
-    ):
-        binary_op_result_column = cast(
-            floor(first_operand / second_operand), LongType()
+
+    def _get_result(self) -> SnowparkPandasColumn:
+        return SnowparkPandasColumn(
+            snowpark_column=self.result_column,
+            snowpark_pandas_type=self.result_snowpark_pandas_type,
         )
-        snowpark_pandas_type = TimedeltaType()
-    elif (
-        op == "mod"
-        and isinstance(first_datatype(), TimedeltaType)
-        and _is_numeric_non_timedelta_type(second_datatype())
-    ):
-        binary_op_result_column = ceil(
-            compute_modulo_between_snowpark_columns(
-                first_operand, first_datatype(), second_operand, second_datatype()
+
+    def _check_timedelta_with_none(self) -> bool:
```

Comment:
I prefer that we remove the side effects from this method and assign `result_column` `result_snowpark_pandas_type`  where we call `_check_timedelta_with_none`.